.github/workflows/verify.yaml

```diff
@@ -138,7 +138,9 @@ jobs:
           - {
               platform: "centos7",
               arch: "amd64",
-              compose_args: "-e NANOARROW_ACCEPT_IMPORT_GPG_KEYS_ERROR=1"
+              # Currently the Python on the centos7 image is 3.6, which does not support
```

Comment:
Instead of using a base centos 7 you might also want to just use the same centos image that cibuildwheel uses for python packaging. That has quite a few versions of python installed on it

quay.io/pypa/manylinux2014_x86_64